language/move-lang/README.md

```diff
@@ -17,15 +17,44 @@ Move source language is an ergonomic language for writing Modules and Scripts th
+
+* **Stricter than Bytecode** The source language often adds additional levels of restrictions. At an expression level, this means no arbitrary manipulation of the stack (only can do so through expressions), and no dead code or unused effects. At a module level, this could mean additional warnings for unused types or un-invocable functions. At a conceptual/program level, this will also mean adding integration for formal verification.
+
+### Secondary Principles
+
+* **Pathway of Learning** Syntax choices and error messages are intended to give a natural flow of learning. For example, some of the choices around expression syntax could be made more familiar, but they would hurt the plug-n-play feeling of the expression based syntax, which might hurt developing a deeper understanding of the language.
+
+* **Aiding Common Community Patterns** As Move becomes more heavily used, common patterns for modules are likely to appear. Move might add new language features to make these patterns easier, clearer, or safer. But, they will not be added if it violates some other key design goal/value of the language.
+
+* **Careful Optimizations** Optimizations are an important developer tool, as they let a programmer write code in a more natural way. However, all of the optimizations performed must be semantic preserving, to prevent any catastrophic exploits or errors from occurring in optimized code. That being said, it is not the primary goal of the Move source language to produce *heavily* optimized code, but it is a nice feature to have.
+
+### Non-Principles
+
+* **Heavy Abstractions** The Move source language does not intend to hide the details of the Move bytecode, this ranges from everything of references to global storage. There might be some abstractions that make interacting with these items easier, but they should always be available in Move at their most basic (bytecode equivalent) level. This does not mean that conveniences such as easy field access or implicit freezing are against the core set of principles, but only that conveniences should not be ambiguous or opaque in how they interact at the bytecode level. Note though, this does not preclude the addition of features to the lanugage, such as access modifiers that that translate to compiler-generated dynamic checks. It is just that it is not an active goal of the language to add on heavy abstractions just for the sake of obscuring bytecode design choices.
```

Comment:
"such as access modifiers that that translate" double 'that'.
I am not sure about the field access part, that seems to refer to a possible feature that I am not sure we will have (well, at least I am not sure we should have)

Reply:
Very confused, we already have it. 

`&x.f.g.` is `&(&copy(x).f).g`
`x.f.g` is `*&(&copy(x).f).g`

Reply:
I got it, I was confused thinking of the possibility to have public fields. 
Good point